### PR TITLE
[Yaml] don't keep internal state between parser runs

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by curry684 — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Comment by fabpot** [comment]
@xabbuh Can I let you merge 2.7 into 2.8/3.2/master?

**Comment by xabbuh** [comment]
@fabpot done

File: src/Symfony/Component/Yaml/Parser.php [hunk 1]
```diff
@@ -64,23 +64,51 @@ public function parse($value, $exceptionOnInvalidType = false, $objectSupport =
...
+        return $data;
+    }
+
+    private function doParse($value, $exceptionOnInvalidType = false, $objectSupport = false, $objectForMap = false)
+    {
         $this->currentLineNb = -1;
         $this->currentLine = '';
         $value = $this->cleanup($value);
         $this->lines = explode("\n", $value);
+        $this->locallySkippedLineNumbers = array();
```
Comment (stof) [inline src/Symfony/Component/Yaml/Parser.php]: these should also be reset at the end of the parsing to release memort

**Pull request merged**